rust/src/io/reader.rs

```diff
@@ -178,14 +183,20 @@ impl FileReader {
             read_struct_from_buf(&tail_bytes.slice(offset..))?
         };
 
-        let (projection, num_columns) = if let Some(m) = manifest {
-            (m.schema.clone(), m.schema.max_field_id().unwrap() + 1)
+        let mut m: Manifest;
```

Comment:
I don't think this line does anything
```suggestion
```

Reply:
this line is required. because the else block needs place to hold the manifest value that outlives the else block.

Let me try to make this look cleaner

Reply:
Oh I see, because the argument manifest is a reference. Odd.

Reply:
Don't worry about it too much then.